libs/server/Storage/Session/ObjectStore/SortedSetGeoOps.cs

```diff
@@ -41,5 +44,160 @@ public GarnetStatus GeoCommands<TObjectContext>(byte[] key, ref ObjectInput inpu
           where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, ObjectInput, GarnetObjectStoreOutput, long, ObjectSessionFunctions, ObjectStoreFunctions, ObjectStoreAllocator>
             => ReadObjectStoreOperationWithOutput(key, ref input, ref objectContext, ref outputFooter);
 
+        /// <summary>
+        /// Geospatial search and store in destination key.
+        /// </summary>
+        /// <typeparam name="TObjectContext"></typeparam>
+        /// <param name="key"></param>
+        /// <param name="input"></param>
+        /// <param name="output"></param>
+        /// <param name="objectContext"></param>
+        /// <returns></returns>
+        public unsafe GarnetStatus GeoSearchStore<TObjectContext>(ArgSlice key, ArgSlice destination, ref ObjectInput input, ref SpanByteAndMemory output, ref TObjectContext objectContext)
+          where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, ObjectInput, GarnetObjectStoreOutput, long, ObjectSessionFunctions, ObjectStoreFunctions, ObjectStoreAllocator>
+        {
+            var createTransaction = false;
+
+            if (txnManager.state != TxnState.Running)
+            {
+                Debug.Assert(txnManager.state == TxnState.None);
+                createTransaction = true;
+                txnManager.SaveKeyEntryToLock(destination, true, LockType.Exclusive);
+                txnManager.SaveKeyEntryToLock(key, true, LockType.Shared);
+                _ = txnManager.Run(true);
+            }
+            var objectStoreLockableContext = txnManager.ObjectStoreLockableContext;
+
+            var isMemory = false;
+            MemoryHandle ptrHandle = default;
+            var ptr = output.SpanByte.ToPointer();
+            var curr = ptr;
+            var end = curr + output.Length;
+
+            try
+            {
+                var isStoreDist = false;
+                Span<ArgSlice> geoSearchParseState = stackalloc ArgSlice[input.parseState.Count + 1]; // Current MaxSize of this stactalloc will be 144 bytes ( 12 (Max Args) x 12 (sizeof ArgSlice) )
+                var currArgIdx = 0;
+                var i = 0;
+                while (i < input.parseState.Count)
+                {
+                    if (!isStoreDist && input.parseState.GetArgSliceByRef(i).ReadOnlySpan.EqualsUpperCaseSpanIgnoringCase(CmdStrings.STOREDIST))
+                    {
+                        isStoreDist = true;
+                        break;
+                    }
+                    else
+                    {
+                        geoSearchParseState[currArgIdx] = input.parseState.GetArgSliceByRef(i);
+                        currArgIdx++;
+                    }
+                    i++;
+                }
+                geoSearchParseState[currArgIdx++] = isStoreDist ? ArgSlice.FromPinnedSpan(CmdStrings.WITHDIST) : ArgSlice.FromPinnedSpan(CmdStrings.WITHHASH);
```

Comment:
Can add this logic in the implementation-level, no need to allocate space for an extra argument

Reply:
I didn't get what you mean by "implementation-level", can you explain it a bit more? "allocate space" is just a 12 byte stackalloc

Reply:
I mean pass the parseState as-is and handle iterating through it in the object implementation.

Reply:
> handle iterating through it in the object implementation.

Does that mean I can't use GEOSEARCH from the API level? I have to created a separate `SortedSetOperation`, also I can't create an operation called GEOSEARCHSTORE because that operation doesn't perform store.  Also, I shouldn't modify the existing `GEOSEARCH` operation to accept `STOREDIST` option because if I do that then the existing `GEOSEARCH` command will treat `STOREDIST` as a valid input. I am not sure what is the right way to "handle iterating through it in the object implementation"

Can you provide the high-level implementation plan for this as your prefer then I can implement it in that manner?

Reply:
Let's just fix the other small comments, I can revise at a later time and tag you as an FYI :)

Reply:
I have changed the implementation so that it does not use additional stackalloc. Still I have not done the logic in "object implementation". Let me know if this is fine